lib/model/message_list.dart

```diff
@@ -151,6 +151,29 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  void maybeUpdateMessageReactions(ReactionEvent event) {
+    final index = findMessageWithId(event.messageId);
+    if (index == -1) {
+      return;
+    }
+
+    final message = messages[index];
+    final reaction = Reaction(
+      emojiName: event.emojiName,
+      emojiCode: event.emojiCode,
+      reactionType: event.reactionType,
+      userId: event.userId,
+    );
+    switch (event.op) {
+      case ReactionOp.add:
+        message.reactions.add(reaction);
+      case ReactionOp.remove:
+        message.reactions.removeWhere((r) => r == reaction);
```

Comment:
Hmm interesting. This differs from what zulip-mobile does (in `messagesReducer.js`):
```js
            reactions: oldMessage.reactions.filter(
              x => !(x.emoji_name === action.emoji_name && x.user_id === action.user_id),
            ),
```

which is to match only on (message and) user ID and emoji name.

The API docs at https://zulip.com/api/get-events#reaction-remove aren't real clear on how to interpret this aspect.

In the server database, the unique key is user, message, reaction type, and emoji code:
```py
class AbstractReaction(AbstractEmoji):
    class Meta:
        abstract = True
        unique_together = ("user_profile", "message", "reaction_type", "emoji_code")
```

(Seems like that should really be message first, then user — without an index that begins with the message, a query that asks for all the reactions on a given message or set of messages is going to be slow. But the order of columns in the key only affects performance, not semantics.)

That suggests that that's what we should key on as well, in order to correctly simulate what we'd expect to see if we fetched the same message from scratch.

And indeed that looks to be what web does too.

So yeah, let's match on user ID, reaction type, and emoji code here. If there's a match on those but the name was different, we should remove it.